src/main/java/io/snyk/jenkins/SnykToHTML.java

```diff
@@ -3,61 +3,68 @@
 import hudson.EnvVars;
 import hudson.FilePath;
 import hudson.Launcher;
-import hudson.Util;
-import hudson.model.TaskListener;
 import hudson.util.ArgumentListBuilder;
 import io.snyk.jenkins.tools.SnykInstallation;
 import io.snyk.jenkins.transform.ReportConverter;
 import jenkins.model.Jenkins;
 
 import java.io.IOException;
+import java.time.Instant;
 
 import static io.snyk.jenkins.config.SnykConstants.SNYK_REPORT_HTML;
 import static io.snyk.jenkins.config.SnykConstants.SNYK_TEST_REPORT_JSON;
 import static java.nio.charset.StandardCharsets.UTF_8;
 
 public class SnykToHTML {
-  public static void generateReport(
+  public static FilePath generateReport(
     SnykContext context,
     SnykInstallation installation
-  ) throws IOException, InterruptedException {
-    TaskListener log = context.getTaskListener();
-    FilePath workspace = context.getWorkspace();
-    Launcher launcher = context.getLauncher();
-    EnvVars env = context.getEnvVars();
-
-    ArgumentListBuilder args = new ArgumentListBuilder();
-
-    FilePath snykReportJson = workspace.child(SNYK_TEST_REPORT_JSON);
-    if (!snykReportJson.exists()) {
-      log.getLogger().println("Snyk report json doesn't exist");
-      return;
-    }
+  ) {
+    try {
+      FilePath workspace = context.getWorkspace();
+      Launcher launcher = context.getLauncher();
+      EnvVars env = context.getEnvVars();
+
+      FilePath snykReportJson = workspace.child(SNYK_TEST_REPORT_JSON);
+      if (!snykReportJson.exists()) {
+        throw new RuntimeException("Snyk Report JSON does not exist.");
+      }
 
-    workspace.child(SNYK_REPORT_HTML).write("", UTF_8.name());
+      FilePath reportPath = workspace.child(getURLSafeDateTime() + "_" + SNYK_REPORT_HTML);
+      reportPath.write("", UTF_8.name());
+
+      ArgumentListBuilder args = new ArgumentListBuilder()
+        .add(installation.getReportExecutable(launcher))
+        .add("-i", SNYK_TEST_REPORT_JSON);
 
-    args.add(installation.getReportExecutable(launcher));
-    args.add("-i", SNYK_TEST_REPORT_JSON, "-o", SNYK_REPORT_HTML);
-    try {
       int exitCode = launcher.launch()
         .cmds(args)
         .envs(env)
+        .stdout(reportPath.write())
         .quiet(true)
         .pwd(workspace)
         .join();
-      boolean success = exitCode == 0;
-      if (!success) {
-        log.getLogger().println("Generating Snyk html report was not successful");
+
+      if (exitCode != 0) {
+        throw new RuntimeException("Exit Code: " + exitCode);
```

Comment:
Maybe the exception should mention that zero exit code was expected?

```suggestion
        throw new RuntimeException("Non-zero Exit Code. Exit Code: " + exitCode);
```

Reply:
I was on the fence on that, I'll change it.